report_app/main/views.py

```diff
@@ -280,104 +228,58 @@ def is_request_correct(the_request):
-        repository.update_data(new_request.json)
-
-
-@main.route("/update_private_repositories", methods=["POST"])
-def update_private_repositories():
-    """Receive data to either add or update the private repo report item in the table
-       within the database
-
-    Returns:
-        N/A: N/A
-    """
-    logger.debug("update_private_repositories()")
-    apply_private_data(request)
-    return ""
-
-
-@main.route("/update_public_repositories", methods=["POST"])
-def update_public_repositories():
-    """Receive data to either add or update the public repo report item in the table
-       within the database
-
-    Returns:
-        N/A: N/A
-    """
-    logger.debug("update_public_repositories()")
-    apply_public_data(request)
-    return ""
+    return jsonify({"message": "GitHub reports updated"}), 200
 
 
+@main.route("/api/v2/compliant-repository/<repository_name>", methods=["GET"])
+# Deprecated API endpoints: These will be removed in a future release
+@main.route("/api/v1/compliant_public_repositories/endpoint/<repository_name>", methods=["GET"])
 @main.route("/api/v1/compliant_public_repositories/<repository_name>", methods=["GET"])
-def compliant_repository(repository_name):
-    """Find a repository in compliant repositories list
+def display_badge_if_compliant(repository_name: str) -> dict:
+    """Display a badge if a repository is considered compliant.
+    Compliance is determined by the status field in the database.
 
-    Args:
-        repository_name (string): name of repository to find
+    Private repositories are not supported and will return a 403 error.
 
-    Returns:
-        dict: json result true if find repository in compliant repositories list
+    Args:
+        repository_name: the name of the repository to check
     """
-    logger.debug("compliant_repository()")
-    repository = Repositories("public")
-    if repository.is_database_ready():
-        compliant_repos = repository.get_compliant_repositories()
-        for compliant_repo in compliant_repos:
-            if compliant_repo.get("name") == repository_name:
-                return {"result": "PASS"}
-    return {"result": "FAIL"}
-
-
-@main.route(
-    "/api/v1/compliant_public_repositories/endpoint/<repository_name>", methods=["GET"]
-)
-def compliant_repository_endpoint(repository_name):
-    """Find a repository in compliant repositories list
+    dynamo_db = ReportDatabase(
+        table_name=os.getenv("DYNAMODB_TABLE_NAME"),
+        access_key=os.getenv("DYNAMODB_ACCESS_KEY_ID"),
+        secret_key=os.getenv("DYNAMODB_SECRET_ACCESS_KEY"),
+        region=os.getenv("DYNAMODB_REGION"),
+    )
 
-    Args:
-        repository_name (string): name of repository to find
+    try:
+        repository = dynamo_db.get_repository_report(repository_name)
+    except KeyError:
+        abort(404)
+    if repository['data']['is_private']:
+        abort(403, "Private repositories are not supported, and %s is private" % repository_name)
```

Comment:
Small one, f-string syntax is typically considered more readable but not really fussed either way 😄

```python
abort(403, f"Private repositories are not supported, and {repository_name} is private")
```